app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -67,7 +70,14 @@ class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
     );
     final List<_AutoMergeQueryResult> queryResults = await _parseQueryData(data, slug.name);
     for (_AutoMergeQueryResult queryResult in queryResults) {
-      if (mergeCount < _kMergeCountPerCycle && queryResult.shouldMerge) {
+      final RegExp commitRegExp = RegExp(_regExpCommitMessage);
+      final RegExpMatch? match = commitRegExp.firstMatch(queryResult.message);
+      String? revertCommitSha;
+      if (match != null) {
+        revertCommitSha = match.namedGroup('revertCommitSha');
+      }
```

Comment:
Github doesn't treat revert PR/commit separately, and there is not a simple mapping between the revert and the to-be-reverted commit.

Checked recent reverts, and all are with the default commit message which includes the to-be-reverted commit sha info. (this is what is proposed here).

Another more stable way is use git diff between the current revert commit and the second TOT commit: if diff is zero, then we can conclude the current revert is  based on the TOT commit. Filed https://github.com/SpinlockLabs/github.dart/issues/283 to request support on git diff comparison.

Reply:
With https://github.com/flutter/flutter/commit/86cb2598a3b683ebb7bdb9ed4ba6bb0b57424d05 as an example,

Cocoon could grab
1. The proposed revert PR diff: https://github.com/flutter/flutter/commit/86cb2598a3b683ebb7bdb9ed4ba6bb0b57424d05.patch
2. The tip of tree commit diff: https://github.com/flutter/flutter/commit/9412edf1ca69121fe733160e22bc48131e7a3fcd.patch

It can then split the lines of each, and for each line that starts with `+` or `-` add it to a map. When going through the second diff, it can reverse the operator and remove it from the map. If the map has any keys remaining, it's not a pure revert and should be skipped.

Reply:
https://docs.github.com/en/rest/reference/commits#compare-two-commits does exactly the same thing. It sounds a re-implementation of this API. Submitted https://github.com/SpinlockLabs/github.dart/pull/285 to support the `files` field so that we can use this API.

This is the diff result of a revert commit: https://github.com/flutter/flutter/pull/94611/commits/9d2bf9140499984aff7d78145029cfc2a39a762d and the second tot commit: 86d229dc399136aa7a0b3935f723ffa1289217f8
https://api.github.com/repos/flutter/flutter/compare/86d229dc399136aa7a0b3935f723ffa1289217f8...9d2bf9140499984aff7d78145029cfc2a39a762d, where the empty `files` field validates this is a clean revert.